ci-operator/step-registry/cucushift/installer/check/ibmcloud/vpc-resourcegroup/cucushift-installer-check-ibmcloud-vpc-resourcegroup-commands.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+function ibmcloud_login {
+    export IBMCLOUD_CLI=ibmcloud
+    export IBMCLOUD_HOME=/output
+    region="${LEASED_RESOURCE}"
+    export region
+    "${IBMCLOUD_CLI}" config --check-version=false
+    echo "Try to login..."
+    "${IBMCLOUD_CLI}" login -r ${region} --apikey @"${CLUSTER_PROFILE_DIR}/ibmcloud-api-key"
+}
+
+#OCP-60816 - [IPI-on-IBMCloud] install cluster under BYON with a different resource group
+function getResources() {
+    "${IBMCLOUD_CLI}" resource service-instances --type all --output JSON | jq -r '.[]|.name+" "+.resource_id' | sort
+}
+
+function checkResourcesInVPC {
+    local preResourceFile provisioned_rg_file resource_group curResourceFile cmpRst
+    preResourceFile="${SHARED_DIR}/vpc_resources"
+    provisioned_rg_file="${SHARED_DIR}/ibmcloud_resource_group"
+    resource_group=$(cat "${provisioned_rg_file}")
+    "${IBMCLOUD_CLI}" target -g ${resource_group}
+    curResourceFile=$(mktemp)
+    getResources > ${curResourceFile}
+    echo "diff ${curResourceFile} ${preResourceFile} ..."
+    cmpRst=$(diff ${curResourceFile} ${preResourceFile})
+    if [ -n "$cmpRst" ]; then
+        echo "ERROR: resources differ: ${cmpRst}"
+        return 1
+    else
+        echo "checkResourcesinVPC PASS"  
+        return 0
+    fi
+}
+
+if [ -f "${SHARED_DIR}/kubeconfig" ] ; then
+    export KUBECONFIG=${SHARED_DIR}/kubeconfig
+else
+    echo "ERROR: fail to get the kubeconfig file under ${SHARED_DIR}!!"
+    exit 1
+fi
+
+if [ -f "${SHARED_DIR}/proxy-conf.sh" ] ; then
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+check_result=0
+
+rg_file="${SHARED_DIR}/ibmcloud_cluster_resource_group"
+if [ "${CREATE_CLUSTER_RESOURCE_GROUP}" == "yes" ]; then
+    if [ -s "${rg_file}" ]; then
+        ibmcloud_login
+        if ! checkResourcesInVPC ; then
```

Comment:
From the content of this script, and compared with test case steps, I think here need to check:
1. vpc resource group and cluster resource group are different.
2. cluster are created in the cluster resource group.

And actually I did not get the point, why we need to check resources in vpc resource group and compare the resources created in provision step.

Another point, the step is named `vpc-resourcegroup`, but I think it is not proper here, I think `resourcegroup` should be better.